sklearn/datasets/tests/test_base.py

```diff
@@ -182,19 +187,24 @@ def test_load_diabetes():
 
     # test return_X_y option
     check_return_X_y(res, partial(load_diabetes))
+    # test as_frame option
+    check_pandas_dependency_message(load_diabetes)
+    check_as_frame(res, partial(load_diabetes))
 
 
 def test_load_linnerud():
     res = load_linnerud()
     assert res.data.shape == (20, 3)
     assert res.target.shape == (20, 3)
-    assert len(res.target_names) == 3
```

Comment:
Why has this changed? The default behavior of the loader should not change with `as_frame=False`, no?